src/IW/Sync/Search.hs

```diff
@@ -1,31 +1,60 @@
-module IW.Server.Search where
+module IW.Sync.Search
+       ( fetchAllHaskellRepos
+       , fetchAllHaskellIssues
+       , fetchOpenIssuesWithLabels
+       ) where
 
-import GitHub (Error, Repo, SearchResult, Issue)
+import GitHub (SearchResult (..))        
 import GitHub.Endpoints.Search (searchRepos, searchIssues)
 
+import IW.App (WithError)
+import IW.Db (WithDb)
+import IW.App.Error (AppErrorType, throwError)
+
+import qualified GitHub (Repo, Issue, Error (..))
+
+
 -- | Fetch all repositories with Haskell language
-fetchAllHaskellRepos :: IO (Either Error (SearchResult Repo))
+fetchAllHaskellRepos :: IO (Either GitHub.Error (SearchResult GitHub.Repo))
 fetchAllHaskellRepos = searchRepos "language:haskell"
 
+githubErrToAppErr :: GitHub.Error -> AppErrorType
+githubErrToAppErr = undefined
+
+syncAllHaskellRepos :: (MonadIO m, WithDb env m, WithError m) => m ()
+syncAllHaskellRepos = do
+    result <- liftIO fetchAllHaskellRepos
+    case result of
+        Right (SearchResult count vec) -> undefined 
+        Left err                       -> throwError . githubErrToAppErr $ err
+        
+{-
 -- | Fetch all repositories with Haskell language and label "help wanted"
 fetchHaskellReposHW :: IO (Either Error (SearchResult Repo))
```

Comment:
Yeah, you can delete these functions, no need to have these two specializations. We have a general function to fetch issues by label.